services/galley/src/Galley/API/Teams.hs

```diff
@@ -427,22 +427,22 @@ updateTeamMember zusr zcon tid targetMember = do
       -- target user must be in same team
       throwM teamMemberNotFound
     Just previousMember -> do
-      okToDowngrade <- canDowngradeTeamMember user previousMember targetPermissions
-      unless okToDowngrade $ throwM accessDenied
+      when (downgradesOwner previousMember targetPermissions)
+        $
+        -- owner can be downgraded IFF it can be deleted
+        unless (canDeleteMember user previousMember)
+        $ throwM accessDenied
```

Comment:
much nicer, thanks!

Reply:
I don't like what ormolu did with the `$`, but I'll leave it now.